tests/unit_tests/ensemble_evaluator/test_monitor.py

```diff
@@ -64,9 +64,11 @@ async def test_unexpected_close(unused_tcp_port):
     )
 
     set_when_done = asyncio.Event()
+    socket_closed = asyncio.Event()
 
     async def mock_ws_event_handler(websocket):
         await websocket.close()
+        socket_closed.set()
```

Comment:
Is it to avoid signal_done() being sent before websocket.close() is finished?

Reply:
> I am not sure I understand what is being tested, but if it is expecting a `ConnectionClosedOK` exception when there is an unexpected `websocket.close()` then it seems to defeat the purpose to inject a `socket_closed` signal in order for that to happen.

I think it makes sense as this will guarantee that the webscoket was closed. We test here that `signal_done` will produce `ConnectionClosedOK` once the ee close the connection.